3party/opening_hours/opening_hours_parsers.hpp

```diff
@@ -0,0 +1,496 @@
+#pragma once
+
+#include "opening_hours.hpp"
+
+// #define BOOST_SPIRIT_DEBUG
+#define BOOST_SPIRIT_USE_PHOENIX_V3
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/phoenix_operator.hpp>
+#include <boost/spirit/include/phoenix_fusion.hpp>
+#include <boost/spirit/include/phoenix_stl.hpp>
+#include <boost/spirit/include/phoenix_statement.hpp>
+#include <boost/spirit/include/phoenix_bind.hpp>
+#include <boost/spirit/include/phoenix_object.hpp>
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshorten-64-to-32"
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/gregorian/gregorian.hpp>
+#pragma clang diagnostic pop
+#else
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <boost/date_time/gregorian/gregorian.hpp>
+#endif
+
+#include "opening_hours_parsers_terminals.hpp"
+
+namespace osmoh
+{
+namespace phx = boost::phoenix;
+
+class test_impl
+{
+public:
+  template <typename T>
+  struct result { typedef void type; };
+
+  template <typename Arg>
+  void operator() (const Arg & a) const
+  {
+    std::cout << a << " \t(" << typeid(a).name() << ")" << std::endl;
+  }
+};
+phx::function<test_impl> const test = test_impl();
```

Comment:
It's a straight way to ODR violation, right? Do we need this variable in header and at the global scope?